### ISSUE TITLE
ERROR in: Now looping and iteratively modify the GFA until all reads align end-to-end on the assembly graph

### ISSUE THREAD

**Issue by octpalacios** [RolandFaure/Hairsplitter#10]
Hi,
I’m using Hairsplitter on a large repetitive genome assembled with Flye, but I encountered the following error:

==== Now looping and iteratively modify the GFA until all reads align end-to-end on the assembly graph ====

 Loop iteration 0... 
  - Going through the gaf file and listing the reads that do not align end-to-end on the assembly graph...
  - Outputting non-duplexed reads in /crex/proj/snic2021-6-151/nobackup/private/Octavio/V.viatia.P24_XY_male_assembly/PacBio_reads/Flye_assembly/Hairsplitter/hairsplitter_out/tmp/reads.fa, discarding in total 87225 reads
  - Pooling the reads that display similar behaviour on the assembly graph...
  - Found 9726 solid bridges and 41160 solid piers.
  - Computing the exact location of new links in the GFA and gap-filling if necessary...
terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::substr: __pos (which is 3792) > this->size() (which is 60)

Do you have any suggestions on how to fix this?
Thanks in advance.

**Comment by RolandFaure** [comment]
Hi,

This is clearly a bug, which may originate from the very repetitive nature of your genome. I need to correct this. If you want a workaround, you can skip step 0 of the pipeline by toggling the --correct-assembly flag, but this will diminish the quality of the result.

To solve this bug, could you show me your command line ? What version of HairSplitter are you using ? And would it be possible to share the files so that I can take a look, or are the files confidential or too big ?

**Comment by RolandFaure** [comment]
Yes, the GFA and the reads would be perfect for me, at roland.faure@irisa.fr. I will delete all the files once the debugging is done. 
Not using --correct-assembly will skip over the buggy part but will decrease the quality of the results.

**Comment by octpalacios** [comment]
Hi Roland,

The files are quite large, and I’m working on a way to send them. In the meantime, I’ll try running Hairsplitter without the "--correct-assembly" flag.

Thanks.

**Comment by RolandFaure** [comment]
Hi, 

If this help, the qualities of the reads in the fastq file are not used, hence a fasta file of the reads is enough. 

Roland
